codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoSettings.java

```diff
@@ -147,4 +152,48 @@ public String getModuleDescription() {
     public void setModuleDescription(String moduleDescription) {
         this.moduleDescription = Objects.requireNonNull(moduleDescription);
     }
+
+    /**
+     * Gets the configured protocol to generate.
+     *
+     * @return Returns the configured protocol.
+     */
+    public ShapeId getProtocol() {
```

Comment:
Probably worth having a method to get the actual trait, as it will contain protocol configuration.